source/simulator/core.cc

```diff
@@ -2430,7 +2465,11 @@ namespace aspect
               goto start_time_iteration;
           }
 
-        postprocess ();
+        // if we use a nonlinear solver scheme and postprocess nonlinear iterations, this function is called within
+        // solve_timestep () in the individual solver schemes
+        if ((!parameters.run_postprocessors_on_nonlinear_iterations) || parameters.nonlinear_solver == NonlinearSolver::IMPES
```

Comment:
Yes, I had thought about that too, and now seeing the changes it's much better than before.